agent/containermetadata/write_metadata_unix.go

```diff
@@ -0,0 +1,83 @@
+// +build !windows
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerclient"
+)
+
+const (
+	mountPoint                     = "/opt/ecs/metadata"
+	ContainerMetadataClientVersion = dockerclient.Version_1_21
+)
+
+// createMetadataFile initializes the metadata file
+func createMetadataFile(metadataDirectoryPath string) error {
+	metadataFilePath := filepath.Join(metadataDirectoryPath, metadataFile)
+	temp, err := ioutil.TempFile(metadataDirectoryPath, "temp_metadata_file")
+	if err != nil {
+		return err
+	}
+	defer temp.Close()
+	err = temp.Chmod(readOnlyPerm)
```

Comment:
In that case, do you think it is advisable to make the permission 0600 instead, rw for only root user? This should prevent arbitrary users outside the container from reading the file, if this is something to be concerned about (users putting sensitive info in some container data?)